pkg/cmd/admin/router/router.go

```diff
@@ -477,7 +479,8 @@ func RunCmdRouter(f *clientcmd.Factory, cmd *cobra.Command, out io.Writer, cfg *
 			return fmt.Errorf("router could not be created; you must specify a service account with --service-account")
 		}
 
-		err := validateServiceAccount(kClient, namespace, cfg.ServiceAccount)
+		var err error
+		accountExists, needsSCC, err = validateServiceAccount(kClient, namespace, cfg.ServiceAccount, cfg.HostNetwork)
```

Comment:
It would but I didn't see them used outside the scope of the `if generate {` code  block. 
You would need to remove that outer assignment to `accountExists, needsSCC := false, false` (should catch an unused error I think).  Or did I read this wrong?